src/hotspot/share/oops/arrayKlass.hpp

```diff
@@ -56,13 +57,13 @@ class ArrayKlass: public Klass {
   int dimension() const                 { return _dimension;      }
   void set_dimension(int dimension)     { _dimension = dimension; }
 
-  Klass* higher_dimension() const     { return _higher_dimension; }
-  inline Klass* higher_dimension_acquire() const; // load with acquire semantics
-  void set_higher_dimension(Klass* k) { _higher_dimension = k; }
-  inline void release_set_higher_dimension(Klass* k); // store with release semantics
+  ObjArrayKlass* higher_dimension() const     { return _higher_dimension; }
+  inline ObjArrayKlass* higher_dimension_acquire() const; // load with acquire semantics
+  void set_higher_dimension(ObjArrayKlass* k) { _higher_dimension = k; }
```

Comment:
The set_higher_dimension() function seems to be unused.  Should we remove it?

Reply:
I'm using the function in the following PR https://github.com/openjdk/jdk/pull/14959.

Reply:
Should you be using release_set_higher_dimension in your change @calvinccheung  ?  It does seem better to not have unsafe alternatives.

Reply:
Sure, I can use release_set_higher_dimension in my change.

Reply:
That would be good.  Not this change, but the higher and lower dimensions have to change to not be set inside a Mutex so it'll be safer to only have the acquire/release versions.

Reply:
Thank you for noticing the set_higher_dimension function is unused, Dean.

Reply:
But Calvin wants it and doesn't need the release version. Using release where not needed just causes confusion because it screams out "something lock-free and concurrent is happening here", which is not the case in Calvin's code.

Reply:
I can add it back and we can examine it when we fix JDK-8308745.  Maybe add an assert or something that the vm is single threaded.